go/vt/binlog/keyrange_filter.go

```diff
@@ -40,40 +30,23 @@ func KeyRangeFilterFunc(kit key.KeyspaceIdType, keyrange *pb.KeyRange, sendReply
 				log.Warningf("Not forwarding DDL: %s", statement.Sql)
 				continue
 			case proto.BL_DML:
-				keyspaceIndex := strings.LastIndex(statement.Sql, KEYSPACE_ID_COMMENT)
-				if keyspaceIndex == -1 {
+				keyspaceId, unfriendly, err := annotations.ParseSQLAnnotation(string(statement.Sql))
+				if err != nil {
```

Comment:
Hmm. I agreed at first, but after thinking more about it, I think we'd like to have a separate case for "unfriendly". We want to know that batch jobs that are not supposed to be running (and generating unfriendly statements) are, rather than just skip statements. I'd also like to change this code to abort the filtered replication, in this case, rather than skipping it. 
What do you think?